src/libbson/src/bson/bson.c

```diff
@@ -3584,7 +3584,19 @@ _bson_iter_validate_document (const bson_iter_t *iter,
       state->phase = BSON_VALIDATE_PHASE_LF_REF_KEY;
    }
 
-   (void) bson_iter_visit_all (&child, &bson_validate_funcs, state);
+   bson_iter_visit_all_v2 (
+      &child, &bson_validate_funcs, BSON_ITER_VISIT_NOFLAGS, state);
+
+   if (child.err_off > 0 && state->err_offset < 0) {
```

Comment:
```suggestion
   if (child.err_off > 0 && state->err_offset <= 0) {
```
Should it be `state->err_offset <= 0` for consistency with the condition used below (line 3604)? Or should the condition below be changed to `state->err_offset < 0`?